library/general/test/report_test.rb

```diff
@@ -0,0 +1,105 @@
+#! /usr/bin/env rspec
+#
+require_relative "test_helper"
+require "yaml"
+
+Yast.import "Report"
+Yast.import "Mode"
+
+describe Yast::Report do
+  before { subject.ClearAll }
+
+  describe ".Warning" do
+    let(:show) { true }
+    let(:message) { "Message" }
+
+    context "while in command-line mode" do
+      before(:each) do
+        allow(Yast::Mode).to receive(:commandline).and_return(true)
+      end
+
+      it "prints the message only on console" do
+        expect(Yast::CommandLine).to receive("Print")
+          .with(/#{message}/)
+        expect(Yast::Popup).to_not receive("Warning")
+        expect(Yast::Popup).to_not receive("TimedWarning")
+        subject.Warning(message)
+      end
+    end
+
+    context "while in UI mode and timeout is disabled" do
+      let(:timeout) { 0 }
+
+      before(:each) do
+        subject.DisplayWarnings(show, timeout)
+        allow(Yast::Mode).to receive(:commandline).and_return(false)
```

Comment:
If you are going to mock `Yast::Mode.commandline` in all cases, maybe you would like to move it to a before block and just define the value for each case.

```ruby
before do
  allow(Yast::Mode).to receive(:commandline).and_return(commandline?)
end

context "while in UI mode and timeout is disabled" do
  let(:timeout) { 0 }
  let(:commandline?) { false }

  it "shows a popup" do
    #...
  end
end
```